app/models/validations/date_validations.rb

```diff
@@ -91,4 +101,15 @@ def date_valid?(question, record)
   def is_rsnvac_first_let?(record)
     [15, 16, 17].include?(record["rsnvac"])
   end
+
+  def inactive_status(date, deactivation_periods, available_from)
+    return if date.blank?
+
+    closest_reactivation = deactivation_periods.order(created_at: :desc).find { |period| period.reactivation_date.present? && date.between?(period.deactivation_date, period.reactivation_date - 1.day) } if deactivation_periods.present?
```

Comment:
```suggestion
    closest_reactivation = deactivation_periods.order(created_at: :desc).find { |period| period.reactivation_date.present? && date.between?(period.deactivation_date, period.reactivation_date - 1.day) } if deactivation_periods.present?
```

Can you make this an ActiveRecord query?